mmdet3d/datasets/once_dataset.py

```diff
@@ -0,0 +1,289 @@
+        """
+        return len(self.data_infos)
+    
+    def _check_annos(self, info):
+        return 'annos' in info
+
+    def get_data_info(self, index):
+        """Get data info according to the given index.
+
+        Args:
+            index (int): Index of the sample data to get.
+
+        Returns:
+            dict: Data information that will be passed to the data
+                preprocessing pipelines. It includes the following keys:
+
+                - sample_idx (str): Sample index.
+                - pts_filename (str): Filename of point clouds.
+                - img_prefix (str): Prefix of image files.
+                - img_filename (str, optional): image filename.
+                - lidar2img (list[np.ndarray], optional): Transformations
+                    from lidar to different cameras.
+                - ann_info (dict): Annotation info.
+        """
+        info = self.data_infos[index]
+        sample_idx = info['frame_id']
+        pts_filename = info['lidar_path']
+        
+        img_filenames = []
+        lidar2imgs = []
+        seq_id = info['sequence_id']
+        for camera in self.camera_list:
+            img_filename = os.path.join(self.data_root, 'data', \
+                                        seq_id, camera, f'{sample_idx}.jpg')
+            img_filenames.append(img_filename)
+            # obtain lidar to image transformation matrix
+            cam2lidar = info['calib'][camera]['cam_to_velo']
+            lidar2cam = np.linalg.inv(cam2lidar)
+            intrinsic = info['calib'][camera]['cam_intrinsic']
+            viewpad = np.eye(4)
+            viewpad[:3, :3] = intrinsic
+            lidar2img = viewpad @ lidar2cam.T
+            lidar2imgs.append(lidar2img)
+
+        input_dict = dict(
+            sample_idx=sample_idx,
+            pts_filename=pts_filename,
+            img_prefix=None,
+            img_filename=img_filenames,
+            lidar2img=lidar2imgs,
+        )
+
+        if not self.test_mode:
+            annos = self.get_ann_info(index)
+            input_dict['ann_info'] = annos
+
+        return input_dict
+
+    def get_ann_info(self, index):
+        """Get annotation info according to the given index.
+
+        Args:
+            index (int): Index of the annotation data to get.
+
+        Returns:
+            dict: annotation information consists of the following keys:
+
+                - gt_bboxes_3d (:obj:`LiDARInstance3DBoxes`):
+                    3D ground truth bboxes.
+                - gt_labels_3d (np.ndarray): Labels of ground truths.
+                - gt_bboxes (np.ndarray): 2D ground truth bboxes.
+                - gt_labels (np.ndarray): Labels of ground truths.
+                - gt_names (list[str]): Class names of ground truths.
+        """
+        # Use index to get the annos, thus the evalhook could also use this api
+        info = self.data_infos[index]
+        annos = info['annos']
+
+        gt_bboxes_3d = annos['boxes_3d']
```

Comment:
Please add a comment here.
\# TODO: convert the box format in the original ONCE coordinate to that in the LiDAR coordinate in mmdet3d.